asserts/store_test.go

```diff
@@ -0,0 +1,119 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts
```

Comment:
Can we test this from outside, naming this as asserts_test?  In my experience this inherently encourages the use of better testing practices, by making it easier to create blackbox tests than whitebox ones, and thus focusing more on the goals we want out of these APIs than how they are implemented. We can still poke below the surface if we really want, via the export_test.go convention, which sits within the asserts package and has the purpose of making private content public to the tests only, but this extra step again means such low-level access has to be designed rather than carelessly abused just because it's so easy to poke.

Makes sense?